cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java

```diff
@@ -67,19 +69,33 @@ public ModelAndView handleRequestInternal(final HttpServletRequest request, fina
         final UserProfile profile = manager.get(true);
         CommonHelper.assertNotNull("profile", profile);
 
-        // bypass approval -> redirect to the application with code
+        // bypass approval -> redirect to the application with code or access token
         if (bypassApprovalService || bypassApprovalParameter != null) {
             final Principal principal = principalFactory.createPrincipal(profile.getId(), profile.getAttributes());
             final Authentication authentication = new OAuthAuthentication(ZonedDateTime.now(), principal);
             final Service service = new OAuthWebApplicationService("" + registeredService.getId(), registeredService.getServiceId());
-            final OAuthCode code = oAuthCodeFactory.create(service, authentication);
-            logger.debug("Generated OAuth code: {}", code);
-            ticketRegistry.addTicket(code);
 
+            final String responseType = request.getParameter(OAuthConstants.RESPONSE_TYPE);
             String callbackUrl = redirectUri;
```

Comment:
I see your point, though, built urls are very different and only used in that class. I'm hesitating on this.